### PR TITLE
Add and refactor schema validation

### PR DESCRIPTION
**Pull request by mojotalantikite**
Last week we were testing a move of some of our Akamai Terraform based code to the new official provider and hit some simple validation issues. This change adds a few simple validation checks and refactors the validateRecord func to a more idiomatic Go version.

Over the past few months I had been working on an internal Terraform provider that targets v2 of the FastDNS API. That code can be found here:
https://github.com/trussworks/internal-terraform-provider-akamai
https://github.com/trussworks/akamai-sdk-go

We'd be happy to start trying to upstream some of that code to your official repositories if your dev teams are open to that. We thought it'd be best to start with this PR before we start pulling our code in and making more PRs. Thanks!

### CODE REVIEW

**Comment by mojotalantikite** [comment]
Looks like Travis failed due to some dependency issues, likely due to using older versions of Go that don't use go modules.

Let me know what your team prefers to do for dependency things, or if you can drop support for Go versions that don't use go modules. Thanks!

**Comment by dshafik** [comment]
@mojotalantikite this is great work!

We are currently committed to using Go modules, so dropping support for older versions is the way to go.

I'd love to get Travis passing before merging this, is that something you want to do, or should I take a stab at it?

**Comment by mojotalantikite** [comment]
@dshafik thanks! I'd be happy to fix the Travis build up to get it to use versions of Go that support Go modules. I'll try to get this passing sometime today.

**Comment by mojotalantikite** [comment]
Ok, so I fixed this up, but a couple things came out of it:

1) go mod did away with vendor/, but since this repo was using it still I didn't want to get rid of it completely in case there was a reason for it. The Travis config I have in there still respects vendor/, but I'd say it probably should be moved away from eventually.

2) As a habit I generally run things through a linter in CI (or, really, in pre-commit) and when I added that in to the repo I found a ton of linter warnings. I ended up turning it off again as there are too many fixes needed to do them in this PR. Happy to help to get those in order at some point in the future though.

TLDR; Travis passes now.

**Comment by dshafik** [comment]
@mojotalantikite this is great work!

HashiCorp requires us to commit the vendor repo, so we need to keep it (we use `go mod vendor` IIRC). Until that policy changes this is the way to go.

I'll go ahead and merge this and the rebase all my PRs against it and see how they fair!

**Comment by dshafik** [comment]
Thanks @mojotalantikite 🙏🏼
